tests/functional/services/policy_engine/conftest.py

```diff
@@ -0,0 +1,386 @@
+            "postgresql://postgres:mysecretpassword@localhost:5432/postgres",
+        )
+
+
+@pytest.fixture(scope="module")
+def anchore_db() -> ContextManager[bool]:
+    """
+    Sets up a db connection to an existing db, and fails if not found/present
+    Different from the fixture in test/fixtures.py in that it does not drop existing data upon making a connection
+    :return: True after connection setup (not actual connection object).
+    :rtype: ContextManager[bool]
+    """
+
+    conn_str = os.getenv("ANCHORE_TEST_DB_URL")
+    assert conn_str
+    config = {"credentials": {"database": {"db_connect": conn_str}}}
+    try:
+        ret = initialize(localconfig=config)
+        yield ret
+    finally:
+        end_session()
+        do_disconnect()
+
+
+def load_seed_file_rows(file_name: str) -> Generator[Dict, None, None]:
+    """
+    Loads database seed files (json lines) and yields the json objects.
+    :param file_name: name of seed file to load
+    :type file_name: str
+    :return: generator yields json
+    :rtype: Generator[Dict, None, None]
+    """
+    json_file = os.path.join(SEED_FILE_DIR, file_name)
+    with open(json_file, "rb") as f:
+        for line in f:
+            linetext = line.decode("unicode_escape").strip()
+            json_content = json.loads(linetext)
+            if file_name in SEED_FILE_TO_METADATA_MAP:
+                json_key = SEED_FILE_TO_METADATA_MAP[file_name]
+                if json_content[json_key] is not None:
+                    json_content[json_key] = json.loads(json_content[json_key])
+            yield json_content
+
+
+def _setup_vuln_data():
+    """
+    Loads database seed files and bulk saves all records direclty to db
+    """
+    with session_scope() as db:
+        all_records = []
+        # set up vulnerability data
+        for seed_file_name, entry_cls in SEED_FILE_TO_DB_TABLE_MAP.items():
+            for db_entry in load_seed_file_rows(seed_file_name):
+                all_records.append(entry_cls(**db_entry))
+        db.bulk_save_objects(all_records)
+        db.flush()
+
+
+def _teardown_vuln_data():
+    """
+    Drops all vulnerability related data
+    """
+    tablenames = [cls.__tablename__ for cls in SEED_FILE_TO_DB_TABLE_MAP.values()]
+    tablenames_joined = ", ".join(map(str, tablenames))
+    engine = get_engine()
+    with engine.connect() as connection:
+        with connection.begin():
+            connection.execute(f"DROP TABLE {tablenames_joined} CASCADE")
+    do_create_tables()
+
+
+@pytest.fixture(scope="module", autouse=True)
+def setup_vuln_data(
+    request,
+    set_env_vars,
+    anchore_db,
+) -> None:
+    """
+    Writes database seed file content to database. This allows us to ensure consistent vulnerability results (regardless of feed sync status).
+    """
+    _teardown_vuln_data()
+    _setup_vuln_data()
+    request.addfinalizer(_teardown_vuln_data)
+
+
+@pytest.fixture
+def clear_database_temporary(request) -> None:
```

Comment:
Named it that because it temporarily clears the database of vuln information and reloads it afterwards. Do you have a suggestion for something else? Maybe `clear_vulns_temporary`?

Reply:
Renamed it to `clear_vuln_data_temporary`

Reply:
renamed again - fixed